nvim-lsp.vim

```diff
@@ -56,13 +60,20 @@ let g:metals_server_version = '0.9.0+18-27d4652a-SNAPSHOT'
       inputBoxProvider             = true;
       quickPickProvider            = true;
       executeClientCommandProvider = true;
+      hoverProvider                = true;
+      decorationProvider           = true;
+      isHttpEnabled                = false; -- Needs to be present
```

Comment:
Yea, I'll look into Metals to see what is happening here. I think I actually may have an idea.

Reply:
Well in the process of looking into this I found another issue. So that's good I guess 😏. We have that we are a `executeClientCommandProvider`, but don't actually have all the cases for all the commands. However, I think that's fine as clients should be able to support a subset of these. The pr in metals firstly just make sure that doctor still works in a client has `isHttpEnabled = true` and also `executeClientCommandProvider = true`.

https://github.com/scalameta/metals/pull/1834

Reply:
Yeah, the http service works fine when enabled. The executeClientCommandProvider was there already from the new file PR. This is initiated from the client, so we can support any subset we want.

Reply:
Alright, I found the issue with Doctor. Once pr I have linked above is merged in, we can go ahead and remove the `isHttpEnabled = false`.

Also, you have a merge conflict since I did add some stuff earlier today.

Reply:
Ok, I've pushed the rebased PR

Reply:
BTW, which font are you using for the g:diagnostic_virtual_text_prefix character in nvim-metals.vim?

Reply:
I actually don't have `g:diagnostic_enable_virtual_text` turned on. When I did I was just using `` as the prefix.